src/processinfo.cpp

```diff
@@ -616,6 +616,23 @@ ProcessInfo::vdsoOffsetMismatch(uint64_t f1, uint64_t f2,
          (f3 != _gettimeofday_offset) || (f4 != _time_offset);
 }
 
+void
+ProcessInfo::addKeyValuePairToCkptHeader(const string &key, const string &value)
```

Comment:
Why is this called `addKeyValuePairToCkptHeader()`?  When I think "CkptHeader", I imagine the header of a DMTCP checkpoint image file?  But if that's the case, where is this function used?

If you'll add new code later that uses it, can you add a comment for this function now, saying how it's supposed to be used?

Reply:
The ProcessInfo object effectively acts as the checkpoint header for DMTCP. It is what dmtcp_restart reads after opening the ckpt image. The Mana plugin will use this API to add information that can be read/interpreted by MANA/MPI-specific code in dmtcp_restart without requiring any other modifications.

Most of the new functions exposed here will be used in the MANA branch, but since these are generic enough, I decided to  create a separate PR for them, just to keep things simpler for the upcoming MANA PR.

Reply:
Thanks.  That's fair.  But we still need a comment to explain this to the casual reader.  I propose:

`// NOTE:  ProcessInfo object acts as the checkpoint header for DMTCP.`